src/_common/content/components/media-upload/media-upload.ts

```diff
@@ -111,6 +106,43 @@ export default class AppContentMediaUpload extends Vue {
 		);
 		if ($payload.success && $payload.mediaItems && $payload.mediaItems.length === 1) {
 			return new MediaItem($payload.mediaItems[0]);
+		} else if ($payload.success === false && $payload.errors.file) {
```

Comment:
`!$payload.success`, I'm not sure if we consistently return false or maybe just not return success at all in some cases.